packages/e2e_tests/cypress/e2e/register-as-an-agent.cy.js

```diff
@@ -1,110 +1,192 @@
-		haveYourSay.findFieldAndEnterText('telephone', '03034445325');
-		projectPage.findAndClickButton('Continue');
-	});
-
-	it('Chooses to represent as A Person and clicks continue', () => {
-		haveYourSay.findAndSelectRadioButton('person');
-		projectPage.findAndClickButton('Continue');
-	});
-
-	it('Enters their full name and clicks continue', () => {
-		haveYourSay.findFieldAndEnterText('full-name', 'Fred Tester');
-		projectPage.findAndClickButton('Continue');
-	});
-
-	it('Confirms they are over 18 and clicks continue', () => {
-		haveYourSay.findAndSelectRadioButton('yes');
-		projectPage.findAndClickButton('Continue');
-	});
-
-	it('Enters their email address and clicks continue', () => {
-		haveYourSay.findFieldAndEnterText('email', 'pinsemail12@examplePINS.com');
-		projectPage.findAndClickButton('Continue');
-	});
-
-	it('Enters their address details and clicks continue', () => {
-		haveYourSay.findFieldAndEnterText('line1', 'TQH');
-		haveYourSay.findFieldAndEnterText('line2', '21 The Square');
-		haveYourSay.findFieldAndEnterText('line3', 'Bristol');
-		haveYourSay.findFieldAndEnterText('postcode', 'BS1 6PN');
-		haveYourSay.findFieldAndEnterText('country', 'United Kingdomw');
-		projectPage.findAndClickButton('Continue');
-	});
-
-	it('Enters their telephone number and clicks continue', () => {
-		haveYourSay.findFieldAndEnterText('telephone', '03334445325');
-		projectPage.findAndClickButton('Continue');
-	});
-
-	it('Adds a comment against a project and clicks continue', () => {
-		haveYourSay.findFieldAndEnterText(
-			'comment',
-			'This is a test comment against the this specific project for automation testing purposes.'
-		);
-		projectPage.findAndClickButton('Continue');
-	});
-
-	it('Declaration', () => {
-		cy.get('a[href*="/agent/declaration"]').click();
-	});
-
-	it('Accept and continue', () => {
-		cy.get('[data-cy="button-accept-and-regoster"]').click();
-	});
-
-	it('User sees the registration complete message', () => {
-		haveYourSay.registrationCompleteText('Registration complete');
+const basePage = new BasePage();
+
+let caseData;
+
+const representationOptions = [
+	{ representation: 'Person', useOrgName: false, confirmationMessage: 'Registration complete' },
+	{
+		representation: 'Organisation',
+		useOrgName: true,
+		confirmationMessage: 'Registration complete'
+	},
+	{ representation: 'Household', useOrgName: false, confirmationMessage: 'Registration complete' }
+];
+
+representationOptions.forEach((representationOptions) => {
+	describe(`Agent registers to have your say on behalf of ${representationOptions.representation}`, () => {
+		before(() => {
+			cy.clearCookies();
+			cy.fixture('caseData').then((data) => {
+				caseData = data;
+				caseData.useOrgName = data.useOrgName;
+			});
+
+			cy.navigateAndSearch('Front Office Auto Test');
+		});
+
+		it('Should click on register to have your say link and assert page content', () => {
+			basePage.clickProjectInformationMenuLink('have-your-say');
```

Comment:
should this be register-have-your-say?

Reply:
with this method, I created a switch case for the side menu links, I called the case 'have-your-say' but its going to 
the register-have-your-say journey. 

`case 'have-your-say':
					cy.get('a[href*="/register-have-your-say"]').click();
					break;`

I can change this in my next PR.